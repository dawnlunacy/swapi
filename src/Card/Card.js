
import React from 'react';
import { Link } from 'react-router-dom';
import './Card.scss'

const Card = ({ data, findCharacters}) => {
    console.log("CardProp",data)
    // console.log("findCharacters", findCharacters)
    // const { date } = new Date(data.release_date + 'T00:00').toString().split(' ').slice(1,4 ).join(' ');
    const { title, episode_id, release_date, name, homeWorld, homeWorldPopulation, species, films } = data;
    // const film = films.map(film => <li> {film }</li>)

    return(
        <div className="card">
            <header className="card-header">
            {/* <h2> {data.title} </h2>
            <h2> {data.name} </h2> */}
            
            { title && <h2> { title } </h2>}
            { name && <h2> { name }</h2> }

            </header>
            <main className="card-main">
            {/* <h3>  {data.species}</h3>
            <h3>  {data.homeWorld}</h3> */}
             {/* <h3>  { species }</h3>
            <h3>  { homeWorld }</h3>
            <h3> { homeWorldPopulation }</h3>
            <h3>  {episode_id}</h3> */}
        {/* <ul> { films.map(film => <li> { film } </li>) } </ul> */}
            <ul> 
            { episode_id && <li> <h3> Episode { episode_id }</h3></li>}
            { release_date && <li><h3> { new Date(data.release_date + 'T00:00').toString().split(' ').slice(1,4 ).join(' ') } </h3></li>}

            { species && <li> Species: {species} </li>}
            { homeWorld && <li> Home World: {homeWorld} </li>}
            { homeWorldPopulation && <li> Population of Home World: {homeWorldPopulation} </li>}
            { species && <li> Species: {species} </li>}
            { films && films.map(film => <li> { film.title } </li>)}
            </ul>

            {/* <h3>{new Date(release_date + 'T00:00').toString().split(' ').slice(1,4 ).join(' ')}</h3> */}
            </main>
            <footer>
            {/* <button onClick={event => findCharacters(data.characters)}> View Characters </button> */}
            <Link to='/movies/moose' className='btn'>  
            { title && <button onClick={event => findCharacters(data.characters)}> View Characters </button> }            
             </Link>
            </footer>
        </div>
        
    )
}

export default Card;



