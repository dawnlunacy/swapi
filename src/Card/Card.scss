.card {
    border: solid 2px black;
    display: block;
    background-color: teal;
}