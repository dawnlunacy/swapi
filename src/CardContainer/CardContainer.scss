.card-container {
    background: black;
    display: grid;
    grid-template-columns: 1fr 1fr;
    justify-items: center;
    grid-row-gap: 25px;
    padding: 30px;
}

header {
    background-color: pink;
    width: 100%;
    color: black;
}
