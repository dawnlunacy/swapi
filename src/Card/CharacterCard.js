
import React from 'react';
import { Link } from 'react-router-dom';
import './Card.scss'

const Card = ({ data, findCharacters}) => {
    // console.log("CardProp",data)
    // console.log("findCharacters", findCharacters)

    return(
        <div className="card">
        <header className="card-header">
        <h2> {data.name} </h2>

        </header>
        <main className="card-main">
        <h3>  {data.species}</h3>

        </main>
        <footer>
        {/* <button onClick={event => findCharacters(data.characters)}> View Characters </button> */}
        <Link to='/movies/moose' className='btn'>  
        <button onClick={event => findCharacters(data.characters)}> View Characters </button>            
         </Link>

        </footer>
    </div>

    )
}

export default Card;