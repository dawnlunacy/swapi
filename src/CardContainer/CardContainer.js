import React from 'react';
import './CardContainer.scss'
import Card from '../Card/Card'
import { Link, NavLink } from 'react-router-dom'

const CardContainer = ({data, getMovieCharacters, currentCharacters, name, quote, level, isLoading, movie}) => {

    const infoCard = data.map(card => {
        return(<Card className="movie-card"
        key={card.episode_id}
        id={card.episode_id}
        data={card}
        findCharacters={getMovieCharacters}
        />)
    })
    return(
        <>
        <header>
            <h1> StarWars</h1>
            { isLoading && movie && 
                <div className="loading-characters">
                    <div className="crawl">
                        <h1> { movie.title } </h1>
                        <p> { movie.release_date } </p>
                        <p> { movie.opening_crawl }</p>
                    </div>
                 </div>}
            <div className="header-display">
            <div className="user-info-display"> 
                <h2>{ name }</h2>
                <h3>{ quote }</h3>
                <h4>{ level }</h4>
            </div>
            <Link to='/'>
            <button className="sign-out-btn">SIGN OUT</button>
            </Link>

            </div>
        </header>
        <nav>
            <NavLink to='/movies'>
            <button className="main-btn">MOVIES</button>
            </NavLink>
            <button className="main-btn">FAVORITES</button>
          </nav>
          
        <div className="card-container">
            { infoCard }
        </div>
        </>
    )
}

export default CardContainer;