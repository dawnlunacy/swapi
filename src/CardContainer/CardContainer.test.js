import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from '../CardContainer/CardContainer'
import Card from '../Card/Card';

describe('CardContainer', () => {
    it('should match the snapshot with all the correct data passing through', () => {
    let mockGetMovieCharacters = jest.fn();
    let data = [{title: 'Attack of the Clones', episode_id: 2, opening_crawl: 'There is unrest in the Galactic...', release_date: '2002-05-16', characters: Array(2)}]
        let name = "Sausage";
        let quote = "I am Spicy";
        let level = "Master";


        const wrapper = shallow(<CardContainer data={data} getMovieCharacters={mockGetMovieCharacters} name={name} quote={quote} level={level}/>)
        expect(wrapper).toMatchSnapshot();
    })
})