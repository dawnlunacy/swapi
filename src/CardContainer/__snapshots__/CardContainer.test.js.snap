// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all the correct data passing through 1`] = `
<Fragment>
  <header>
    <h1>
       StarWars
    </h1>
    <div
      className="header-display"
    >
      <div
        className="user-info-display"
      >
        <h2>
          Sausage
        </h2>
        <h3>
          I am Spicy
        </h3>
        <h4>
          Master
        </h4>
      </div>
      <Link
        to="/"
      >
        <button
          className="sign-out-btn"
        >
          SIGN OUT
        </button>
      </Link>
    </div>
  </header>
  <nav>
    <NavLink
      to="/movies"
    >
      <button
        className="main-btn"
      >
        MOVIES
      </button>
    </NavLink>
    <button
      className="main-btn"
    >
      FAVORITES
    </button>
  </nav>
  <div
    className="card-container"
  >
    <Card
      className="movie-card"
      data={
        Object {
          "characters": Array [
            undefined,
            undefined,
          ],
          "episode_id": 2,
          "opening_crawl": "There is unrest in the Galactic...",
          "release_date": "2002-05-16",
          "title": "Attack of the Clones",
        }
      }
      findCharacters={[MockFunction]}
      id={2}
      key="2"
    />
  </div>
</Fragment>
`;
